src/ngraph/pass/opset0_downgrade.cpp

```diff
@@ -835,15 +850,27 @@ namespace
             make_shared<op::v0::Split>(node->input_value(0), node->input_value(1), splits_unsigned);
 
         replace_node(node, replacement_node);
-        return true;
+        return replacement_node;
     }
 
     using DispatchMap = map<NodeTypeInfo, std::function<bool(shared_ptr<Node> node)>>;
 
     template <typename T>
     bool op_cast_thunk(shared_ptr<Node> node)
     {
-        return op_cast(as_type_ptr<T>(node));
+        auto downgraded_node = op_cast(as_type_ptr<T>(node));
+        if (downgraded_node)
+        {
+            if (ngraph::get_provenance_enabled())
+            {
+                const std::string provenance_tag =
+                    "<Opset0_Downgrade (v1 " + std::string(node->get_type_name()) + ")>";
+                downgraded_node->add_provenance_tag(provenance_tag);
```

Comment:
since `provenance_tag` for this node will be added in `add_provenance_tags_above` method this line is unnecesary.
```suggestion
```

Reply:
Applied